packages/superset-ui-plugin-chart-word-cloud/src/WordCloud.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { scaleLinear } from 'd3-scale';
+import cloudLayout, { Word } from 'd3-cloud';
+import { CategoricalColorNamespace } from '@superset-ui/color';
+
+const ROTATION = {
+  flat: () => 0,
+  /* eslint-disable-next-line no-magic-numbers */
+  random: () => Math.floor(Math.random() * 6 - 3) * 30,
+  /* eslint-disable-next-line no-magic-numbers */
+  square: () => Math.floor(Math.random() * 2) * 90,
+};
+
+interface Datum {
+  size: number;
+  text: string;
+}
+
+export interface Props {
+  colorScheme: string;
+  data: Datum[];
+  height: number;
+  rotation: 'flat' | 'random' | 'square';
+  sizeRange: number[];
+  width: number;
+}
+
+interface State {
+  words: Word[];
+}
+
+export default class WordCloud extends React.PureComponent<Props, State> {
+  isMounted: boolean = false;
+  state: State = {
+    words: [],
+  };
+
+  componentDidMount() {
+    this.isMounted = true;
+    this.update();
+  }
+
+  componentDidUpdate() {
+    this.update();
+  }
+
+  componentWillUnmount() {
+    this.isMounted = false;
+  }
+
+  update() {
```

Comment:
From the story book, it seems that this method is running forever and in the end there is some console log error.

Reply:
Oh... Will check